### PR TITLE
Adding uploading image in model

### PR DESCRIPTION
**Pull request by capripio**
This PR allow user to use Image Field from standard fields in backpack

### CODE REVIEW

**Comment by tabacitu** [comment]
Hi @capripio ,

Thanks for the PR, looks great. I'll test and merge it ASAP.

Cheers!

File: src/app/Models/Setting.php [hunk 1]
```diff
@@ -4,11 +4,34 @@
...
 
 class Setting extends Model
 {
     use CrudTrait;
 
     protected $table = 'settings';
     protected $fillable = ['value'];
+
+
+    public function setValueAttribute($value)
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: PHP Doc!

File: src/app/Models/Setting.php [hunk 2]
```diff
@@ -4,11 +4,34 @@
...
+    {
+        $attribute_name = "value";
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: If that `$value` above is the attribute name, why not call it attribute name and refactor as appropriate?

File: src/app/Models/Setting.php [hunk 3]
```diff
@@ -4,11 +4,34 @@
...
+        $disk = "public";
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Why isn't this configurable? And why default to "public"?

File: src/app/Models/Setting.php [hunk 4]
```diff
@@ -4,11 +4,34 @@
...
+        $destination_path = $this->key;
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Is key really unique for EVERYTHING in the (relevant, probably `public`) disk?

File: src/app/Models/Setting.php [hunk 5]
```diff
@@ -4,11 +4,34 @@
...
+        if ($value==null) {
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Errrk.

`is_null`

File: src/app/Models/Setting.php [hunk 6]
```diff
@@ -4,11 +4,34 @@
...
+            Storage::disk($disk)->delete($this->image);
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Should one/someone check the return of this? I forget if a `Storage::delete` throws an exception of not if a delete fails.

File: src/app/Models/Setting.php [hunk 7]
```diff
@@ -4,11 +4,34 @@
...
+            $this->attributes[$attribute_name] = null;
+        }
+        if (starts_with($value, 'data:image'))
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Space above.

File: src/app/Models/Setting.php [hunk 8]
```diff
@@ -4,11 +4,34 @@
...
+        {
+            $image = Image::make($value);
+            $filename = md5($value.time()).'.jpg';
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Why `.jpg`? (I'm more being a devil's advocate here).

File: src/app/Models/Setting.php [hunk 9]
```diff
@@ -4,11 +4,34 @@
...
+            Storage::disk($disk)->put($destination_path.'/'.$filename, $image->stream());
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: What if that fails?

File: src/app/Models/Setting.php [hunk 10]
```diff
@@ -4,11 +4,34 @@
...
+        $attribute_name = "value";
+        $disk = "public";
+        $destination_path = $this->key;
+        if ($value==null) {
+            Storage::disk($disk)->delete($this->image);
+            $this->attributes[$attribute_name] = null;
+        }
+        if (starts_with($value, 'data:image'))
+        {
+            $image = Image::make($value);
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Is `Image::make()` always guaranteed to work?

File: src/app/Models/Setting.php [hunk 11]
```diff
@@ -4,11 +4,34 @@
...
+            $filename = md5($value.time()).'.jpg';
+            Storage::disk($disk)->put($destination_path.'/'.$filename, $image->stream());
+            $this->attributes[$attribute_name] = $destination_path.'/'.$filename;
+        }else{
```
Comment (lloy0076) [inline src/app/Models/Setting.php]: Odd spacing.

**Comment by lloy0076** [comment]
@capripio - any updates on this?

**Comment by lloy0076** [comment]
@tabacitu @capripio - ok what's up with this? 😈

**Comment by tabacitu** [comment]
@lloy0076 , @capripio - sorry, I still like this. I think it's a good change to Settings, just haven't been able to dedicate too much time to Settings lately. Will do this week.

**Pull request closed without merging**